narwhal/primary/src/proposer.rs

```diff
@@ -546,6 +551,14 @@ impl Proposer {
                             let _ = self.tx_narwhal_round_updates.send(self.round);
                             self.last_parents = parents;
 
+                            // we re-calculate the timeout to give the opportunity to the node
+                            // to propose earlier if it's a leader for the round
+                            // Reschedule the timer.
+                            let deadline = self.timeout_value();
+
+                            if timer.deadline() > deadline {
```

Comment:
I wonder if we should just reset the timer regardless.

Reply:
Yes I think we should (not sure it changes much but it feels cleaner)